### PR TITLE
FV variable support for ThermochimicaNodalData

### LINKED ISSUE #25661: Allow ThermochimicaNodalData UO to write to FV variables
## Reason
Apps like Pronghorn use FV variables but for coupled applications such as source term analyses for MSR, data from Thermochimica might be needed which is currently only written to FE aux variables. This adds a need for projecting FV elemental values to `MooseVariable`. Extending `ThermochimicaNodalData` UO to write to `MOOSEVariableFVReal` types will eliminate the need for mapping FV data to FE nodes and reduce associated discrepancies. 

## Design
- [x] Modify `Couplable::writableVariable` to return a reference to `MooseVariableField<Real>` type. This is the common parent class for both `MooseVariable` and `MooseVariableFVReal` types.
- [x] Modify `ChemicalCompositionAction` to create FE/FV type variables depending on user choice.

## Impact
Extend usability of `ThermochimicaNodalData` UO.

### PR DESCRIPTION
**Pull request by parikshitbajpai**
Closes #25661
Requires https://github.com/ORNL-CEES/thermochimica/pull/149

### CODE REVIEW

File: modules/chemical_reactions/include/actions/ChemicalCompositionAction.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 #pragma once
 
+#include "AddVariableAction.h"
```
[the remark this answers is not in the document]
Comment (dschwen) [reply modules/chemical_reactions/include/actions/ChemicalCompositionAction.h:12]: Just move this to the .C file

**Review by dschwen — COMMENTED** [review]
Looks fine to me, but coverage is an issue. This might be tough to address, as we are intercepting Thermochimica errors that we do not know how to trigger in the first place.

**Review by lindsayad — CHANGES_REQUESTED** [review]
This looks good to me as well

File: test/tests/auxkernels/nodal_aux_var/tests [hunk 2]
```diff
@@ -29,8 +29,7 @@
     input = 'multi_update_var_test.i'
     exodiff = 'out_multi_var.e'
     issues = '#2099'
-    requirement = "The AuxKernel objects shall be capable of wriyting to to multiple coupled "
-                  "variables."
+    requirement = "The AuxKernel objects shall be capable of writing to multiple coupled variables."
```
Comment (GiudGiud) [inline test/tests/auxkernels/nodal_aux_var/tests:32]:
```suggestion
    requirement = "Auxiliary kernel objects shall be capable of writing to multiple coupled variables."
```

**Comment by parikshitbajpai** [comment]
> Looks fine to me, but coverage is an issue. This might be tough to address, as we are intercepting Thermochimica errors that we do not know how to trigger in the first place.

I figured out a way to trigger those errors and that should increase coverage. Waiting on https://github.com/ORNL-CEES/thermochimica/pull/149 before I push as I'll need to update Thermochimica submodule.

**Comment by GiudGiud** [comment]
when you squash the two commits please add something to the message
some classics : fix typos / adapt tests requirement / add more constness

we try to keep the commit history very informative

**Review by lindsayad — APPROVED** [review]

**Pull request merged**